src/runtime_src/core/tools/xbutil2/EventTraceConfig.cpp

```diff
@@ -0,0 +1,460 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+#include "EventTraceConfig.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iomanip>
+#include <regex>
+#include <set>
+#include <sstream>
+
+#include "core/common/json/nlohmann/json.hpp"
+
+namespace {
+constexpr uint32_t EVENT_BITS_DEFAULT = 16;
+constexpr uint32_t PAYLOAD_BITS_DEFAULT = 48;
+}
+
+event_trace_config::
+event_trace_config(const std::string& json_file_path)
+  : event_bits(EVENT_BITS_DEFAULT),
+    payload_bits(PAYLOAD_BITS_DEFAULT),
+    file_major(0),
+    file_minor(0),
+    code_tables{},
+    category_map{},
+    arg_templates{},
+    event_map{},
+    config_valid(false)
+{
+  load_from_json(json_file_path);
+}
+
+bool 
+event_trace_config::
+load_from_json(const std::string& json_file_path) 
+{
+  try {
+    nlohmann::json config = load_json_file(json_file_path);
+    parse_version(config);
+    parse_data_format(config);
+    parse_lookups(config);
+    parse_categories(config);
+    parse_arg_sets(config);
+    parse_events(config);
+    config_valid = true;
+    return true;
+  } 
+  catch (const std::exception& e) {
+    throw std::runtime_error("JSON parsing error: " + std::string(e.what()));
+  }
+}
+
+nlohmann::json 
+event_trace_config::load_json_file(const std::string& json_file_path) 
```

Comment:
This looks like a general function not specific to the class.  I would move to unnamed namespace static function.